tensorflow_addons/optimizers/lamb_optimizer_test.py

```diff
@@ -0,0 +1,461 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for LAMBOptimizer."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+from numpy import linalg
+
+import tensorflow as tf
+
+from tensorflow_addons.optimizers import lamb_optimizer
+from tensorflow_addons.utils import test_utils
+
+
+def lamb_update_numpy(param,
+                      g_t,
+                      t,
+                      m,
+                      v,
+                      lr=0.001,
+                      lamb_wd=0.0,
+                      beta1=0.9,
+                      beta2=0.999,
+                      epsilon=1e-6):
+
+    m_t = beta1 * m + (1 - beta1) * g_t
+    v_t = beta2 * v + (1 - beta2) * g_t * g_t
+
+    m_t_hat = m_t / (1 - beta1**(t + 1))
+    v_t_hat = v_t / (1 - beta2**(t + 1))
+    update = m_t_hat / (np.sqrt(v_t_hat) + epsilon)
+
+    update += lamb_wd * param
+
+    w_norm = linalg.norm(param, ord=2)
+    g_norm = linalg.norm(update, ord=2)
+    ratio = np.where(w_norm > 0, np.where(g_norm > 0, (w_norm / g_norm), 1.0),
+                     1.0)
+
+    param_t = param - ratio * lr * update
+    return param_t, m_t, v_t
+
+
+def get_beta_accumulators(opt, dtype):
+    local_step = tf.cast(opt.iterations + 1, dtype)
+    beta_1_t = tf.cast(opt._get_hyper("beta_1"), dtype)
+    beta_1_power = tf.math.pow(beta_1_t, local_step)
+    beta_2_t = tf.cast(opt._get_hyper("beta_2"), dtype)
+    beta_2_power = tf.math.pow(beta_2_t, local_step)
+    return (beta_1_power, beta_2_power)
+
+
+class LambOptimizerTest(tf.test.TestCase):
+    @test_utils.run_in_graph_and_eager_modes(use_gpu=False)
```

Comment:
Could you run all tests by decorating the class with `@test_utils.run_all_in_graph_and_eager_modes`